app/src/main/java/com/futurice/freesound/feature/search/SearchActivity.java

```diff
@@ -75,55 +84,62 @@ public void unbind() {
         }
     };
 
-    private void setClearSearchVisible(@NonNull final Boolean clearVisible) {
-        Preconditions.checkNotNull(closeButton, "Close Button has not been bound");
+    private void setClearSearchVisible(final boolean clearVisible) {
+        checkNotNull(closeButton, "Close Button has not been bound");
 
         get(closeButton).setVisibility(clearVisible ? View.VISIBLE : View.GONE);
```

Comment:
I would keep the `checkNotNull` here because it's more informative. Perhaps we could consider a similar utility `checkIsBound(View v)` which might give us more context to such `NPE`s with the burden of the long worded text. Then again, we might just be better off using ButterKnife.